cpp/src/arrow/array.h

```diff
@@ -522,6 +522,44 @@ class ARROW_EXPORT ListArray : public Array {
   std::shared_ptr<Array> values_;
 };
 
+// ----------------------------------------------------------------------
+// FixedSizeListArray
+
+/// Concrete Array class for list data
+class ARROW_EXPORT FixedSizeListArray : public Array {
+ public:
+  using TypeClass = FixedSizeListType;
+
+  explicit FixedSizeListArray(const std::shared_ptr<ArrayData>& data);
+
+  FixedSizeListArray(const std::shared_ptr<DataType>& type, int64_t length,
+                     const std::shared_ptr<Buffer>& value_offsets,
```

Comment:
Why is this constructor taking a value offsets buffer?

Reply:
Good catch